federation/cmd/federation-apiserver/app/server.go

```diff
@@ -48,8 +48,8 @@ import (
 	"k8s.io/kubernetes/pkg/generated/openapi"
 	"k8s.io/kubernetes/pkg/kubeapiserver"
 	kubeapiserveradmission "k8s.io/kubernetes/pkg/kubeapiserver/admission"
-	kubeoptions "k8s.io/kubernetes/pkg/kubeapiserver/options"
-	kubeserver "k8s.io/kubernetes/pkg/kubeapiserver/server"
```

Comment:
Thanks @deads2k for the clarification.

No one will deploy an insecure server in any serious deployments, but I'm afraid there are still many real life cases where people prefer a simpler setup for reasons such as education, integration with other system, etc. (ref: #49933 )

So it seems to me there are several options for decoupling the federation api out:
 - If we want to keep the logic,  we are about to move it to apiserver for sharing. Sounds like this isn't your preference.
 - If we want to delete the support to insecure serving, then delete it from kubernetes code.
-  If federation API wants to support insecure serving, build its own implementation instead of referencing the existing logic.

Any suggestions on moving forward?